cpp/tests/join/join_tests.cpp

```diff
@@ -56,6 +127,13 @@ TEST_F(JoinTest, InvalidCommonColumnIndices)
   Table t1(std::move(cols1));
 
   EXPECT_THROW(cudf::inner_join(t0, t1, {0, 1}, {0, 1}, {{0, 1}, {1, 0}}), cudf::logic_error);
+
+  EXPECT_THROW((build_once_probe_multiple_parallel<join_kind::INNER, false>(
```

Comment:
The OO interface should have it's own tests and not be part of the existing tests. The existing tests will already exercise the OO interface to make sure it preserves existing functionality. Any new functionality (such as probing with multiple tables) should be new tests.

Reply:
Yes, will do.

Reply:
I'm thinking about adding new tests of probing multiple/single table in parallel, and specifying probe output side. These are the new functionalities I can think of. Any other things that I may have missed?

And one more question, should I add tests for all the corner cases that existing tests have already covered? Such as empty table on either/both sides, null equality, etc.

Reply:
> should I add tests for all the corner cases that existing tests have already covered? Such as empty table on either/both sides, null equality, etc.

Since you've refactored the existing public Join APIs to use the OO interface, I think the existing tests are sufficient to exercise this behavior.

I think the only new tests needed are for exercising the new behavior, i.e., probing a single table with multiple other tables. 

I don't think we need to provide tests for probing in parallel. libcudf makes very weak guarantees about thread safety. So the multi-probe tests should just sequentially probe the `hash_join` object with multiple probe tables.

Reply:
Yes, will do.

Reply:
Addressed.